src/Http/Environment.php

```diff
@@ -322,10 +322,14 @@ protected function detectCli(bool|null $override = null): bool
 		}
 
 		// @codeCoverageIgnoreStart
+		if (php_sapi_name() === 'cli') {
+			return true;
+		}
+
 		$term = getenv('TERM');
 
 		if (
-			substr(PHP_SAPI, 0, 3) === 'cgi' &&
+			substr(php_sapi_name(), 0, 3) === 'cgi' &&
```

Comment:
To avoid calling the function twice, we could store `$sapi = php_sapi_name()` first